### ISSUE TITLE
S4 not detected

### ISSUE THREAD

**Issue by svenmeier** [svenmeier/coxswain#33]
Users have reported Coxswain not detecting a connected S4.
At the moment it is not clear what might be causing this:
- the problem seems to be device independent (Z3, Samsung S6, HTC One M7; Huawei P8)
- the S4 connects to a PC without problem
- the OTG adapter can be used to read from USB device storage

**Comment by svenmeier** [comment]
Might be related:
http://stackoverflow.com/questions/33578132/why-cant-this-specific-device-be-enumerated-by-usbmanager-getdeviceslist

**Comment by svenmeier** [comment]
Maybe the power output of the Android device is too low, and an external power supply would help:

http://stackoverflow.com/questions/22202039/do-android-devices-need-to-supply-power-over-usb-in-order-to-support-custom-usb

**Comment by charlespope** [comment]
I have just upgraded my Nexus 5 to the Nexus 6P running Android 7.0. Bought a new OTG adapter for the USB Type-C port and Coxwain cannot detect the rower. Not sure if it is related to the Nexus 6P or the USB Type-C or the Android 7.0 change. Love the app and would like to get it working again.

**Comment by svenmeier** [comment]
Many thanks for chiming in. So your Nexus 5 was connecting fine to the same S4? Do you have a link to the OTG adapter you're using now?

I'll have the opportunity to test a Nexus 5X on Friday (Android 7.0 and USB Type-C), so maybe I'm able to reproduce the problem.

**Comment by charlespope** [comment]
Yes the N5 has been working perfectly.
This is the OTG adapter:
https://www.pbtech.co.nz/product/ADPAPA0611W/Apacer-USB-Type-C-Adapter-White-Dual-Interface-of

Thanks Sven - I appreciate the effort you go to supporting your app.


Charles Pope
Ph: +64 (21) 2727673

**Comment by svenmeier** [comment]
I've found numerous reports of OTG adapters not working with different devices, including the 6P:

http://forum.xda-developers.com/nexus-6p/help/otg-problems-t3266787

At the moment I suspect the adapter to be faulty, but let's wait what for my tests on Friday.

**Comment by svenmeier** [comment]
Nexus 5x (USB-C, Android 7.0) works perfectly fine with the following OTG adapter:

  https://www.amazon.de/Anker-Weiblich-Verwendet-Technologie-ChromeBook/dp/B01COOQIKU

At the moment I can only assume that some OTG adapters are faulty, so maybe yours is too.
You would do me a great favor, if you tried another high-quality adapter. E.g. the following looks similar to mine:

   http://www.trademe.co.nz/mobile-phones/accessories/cables/auction-1189589226.htm

**Comment by charlespope** [comment]
I have ordered the one sold by Google from the Play store. Will let you
know when it arrives in a few weeks.

**Comment by svenmeier** [comment]
Great, please keep us updated.

**Comment by charlespope** [comment]
Hi Sven
My OTG connector arrived - the one sold in the Google store. The Nexux 6P
now detects Coxwain and will launch the program when I connect. With my
cheap OTG connector this did not happen.
However, the Coxwain app does not receive any data from the S4 and the
phone does not show under Settings>Devices.


Charles Pope
Ph: +64 (21) 2727673

**Comment by svenmeier** [comment]
Hi Charles,
many thanks for trying out the new adapter - it seems we have solved the hardware problem at least.

>6P now detects Coxwain and will launch the program when I connect

When you connect the adapter with the S4 already turned on ... it a confirmation dialog asking you whether you want Coxswain to be started? Have you checked "always" on this dialog?

Please do another test: first connect the adapter to your *sleeping* S4 and your 6P, then turn your S4 on. Is Coxswain started?

**Comment by charlespope** [comment]
Yes Coxwain is started whan I connect to the sleeping S4 - and the S4 gets
turned on automatically when I connect.
That did not happen with the old adapter.

Charles Pope
Ph: +64 (21) 2727673

**Comment by svenmeier** [comment]
Does the Coxswain notification appear in the Android status bar, once you connect the Waterrower?

Please activate the hardware trace in Coxswain's settings (hardware/trace), then reconnect the S4. You should find a file "/coxswain/waterrower.trace" on your device, which might give us a clue.

**Comment by charlespope** [comment]
Yes the notification does appear.
I have set the trace on, connected, rowed a few strokes then exited the
app. There is no  "/coxswain/waterrower.trace" file. The directory is also
not there. (Using Asus file manager app).
Hope that helps.

**Comment by svenmeier** [comment]
Ok, let's take a step back. Please verify that

- you can read the contents of an arbitrary USB stick via the new adapter
- the Coxswain icon shows up in the notification area when you connect the S4, and it stays visible *until* you disconnect the S4, when it disappears
- the device is listed in Coxswain's settings under hardware/devices while the Coxswain icon is visible in the notification area

**Comment by charlespope** [comment]
1. Yes I can read a USB stick
2. The Coxwain icon does show up in the notification area when I connect
and it stays visible and disappears when I disconnect - however, this
behavior seems to be intermittent. While connected and while I am holding
my nexus 6p the device keeps vibrating like it is making the connection -
seems to be connected to moving the phone. When I put the phone down it
stops. This may be related to the 6p's ability to "wake up" when you move
it. Sometimes the icon doesn't appear when I connect - I have to move the
phone or disconnect and reconnect - almost like a bad connection - hmmm I
wonder if the cable has a problem - I will try my previous phone (nexus 5)
again which never had any problems (will take a few days to get it back). I
also noticed that sometimes the icon did not disappear when I disconnected.
3. Yes the device is listed in Coxswain's settings under hardware/devices
while the Coxswain icon is visible in the notification area.
[image: Inline images 1]

Android N also has this option when I connect :
[image: Inline images 2]

Hope that helps


Charles Pope
Ph: +64 (21) 2727673

**Comment by svenmeier** [comment]
(2) really looks like a hardware issue. What's strange is that your USB stick works without problems though :/.

**Comment by charlespope** [comment]
Further to my last email - I have been able to get Coxwain to behave
correctly including showing notification icon, receive data from the S4,
remove notification icon on disconnect. This worked when I left the Nexus
P6 firmly on the floor and did not lift or shake it.
When I move the phone it seems to interfere with the connection - the icon
can disappear from notification, no data is received, and on disconnect the
icon remains in the notification bar.
Very confusing and difficult to pin down but seems to be related either to
a faulty cable between the OTG adapter and the S4 or something to do with
the 6P motion sensors. I do not have a second cable but I will test the
cable on my old Nexus 5 which never displayed any issue when I was using
it. Have to get it running again because I did a factory reset.

Charles Pope
Ph: +64 (21) 2727673

**Comment by svenmeier** [comment]
Another test you could perform:
Attach your USB stick via an extension cable (preferably the same length as the Waterrower cable) to the new adapter - it would be interesting whether this causes problems too.
Perhaps the 6P power output on its USB port is too low?

**Comment by svenmeier** [comment]
Closing this for now, until we new evidence of a software problem.

I've noticed rare connection problems with my S4, but could narrow it down to a wonky linkage between the USB cable and the S4 - either the connector on the cable or on the S4 might have loosened.

**Comment by charlespope** [comment]
Hi Sven

Just to close this off - my problem was two fold:
1. the "good quality" Apacer OTG USB Type C adapter
<https://www.pbtech.co.nz/product/ADPAPA0611W/Apacer-USB-Type-C-Adapter-White-Dual-Interface-of>
did not work with my Nexus 6P. When I purchased the OTG USB Type 2 adapter
from the Google Store that problem went away.
2. My intermittent problems connecting to the S4 went away when I replaced
the USB cable.

Thanks for all the suggestions and for spending the time when in fact it
was not a software problem in the end.
I hope this bit of information can help someone else.

cheers
Charlie


Charles Pope
Ph: +64 (21) 2727673
